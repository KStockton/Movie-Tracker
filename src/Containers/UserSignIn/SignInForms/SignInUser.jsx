import React from "react";
import { Redirect } from "react-router-dom";

export const SignInUser = (props) => {
  return props.status === "Successful" ? (
    <Redirect to="/" />
  ) : (
    <section className="user-sign-in-container">
     <div className="user-sign-in-background">
        <img
          alt="movies"
          src="https://i.pinimg.com/originals/86/f1/99/86f199d0c6f5cedd14c121164fa5fafc.jpg"
        />
      </div>
      <form onSubmit={props.handleLogin} className="sign-in-form login-form">
        <h3 className="user-form-title">Sign In</h3>
        <div className="input-container">
          <input
            name="email"
            placeholder="Email..."
            type="email"
            onChange={props.handleChange}
            value={props.email}
            className="user-input"
            id="emails"
          />
        </div>
        <div className="input-container">
          <input
            name="password"
            type="password"
            placeholder="Password..."
            onChange={props.handleChange}
            value={props.password}
            className="user-input"
          />
        </div>
        <span className="status-msg">{props.status}</span>
        <div className="button-container">
          <button
            className="user-form-button sign-in-button"
            onClick={props.handleLogin}
          >
            Sign In
          </button>
          <button className="user-form-link" onClick={props.handleToggleForm}>
            New user? Create an Account!
          </button>
        </div>
      </form>
    </section>
  );
}
