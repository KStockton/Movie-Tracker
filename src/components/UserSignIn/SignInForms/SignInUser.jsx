import React from "react";
import { Redirect } from "react-router-dom";

export default function SignInUser(props) {
  return props.status === "Successful" ? (
    <Redirect to="/" />
  ) : (
    <section className="user-sign-in-container">
      <form onSubmit={props.handleLogin} className="sign-in-form login-form">
        <label>Email</label>
        <input
          name="email"
          type="email"
          onChange={props.handleChange}
          value={props.email}
        />
        <label>Password</label>
        <input
          name="password"
          type="password"
          onChange={props.handleChange}
          value={props.password}
        />
        <button onClick={props.handleLogin}>Sign In</button>
        <button onClick={props.handleToggleForm}>
          New user? Create an Account!
        </button>
      </form>
    </section>
  );
}
