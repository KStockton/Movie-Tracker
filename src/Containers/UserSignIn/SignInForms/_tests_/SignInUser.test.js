import React from "react";
import { shallow, mount } from "enzyme";
import { SignInUser } from "../SignInUser";

const mockhandletoggleform = jest.fn()
const mockhandleLogin =jest.fn()
const mockChange = "cool"

describe("NewUser", () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<SignInUser handleToggleForm={mockhandletoggleform} handleLogin={mockhandleLogin} handleChange={mockChange}/>);
    });
  
     it("NewUser component should match the snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    })
    it('should call handleToggleform when button is clicked', () => {
      wrapper.find(".user-form-link").simulate('click', () => {
        expect(mockhandletoggleform).toHaveBeenCalled()
      })
    })

    it('should call handlelogin when button is clicked', () => {
      wrapper.find(".user-form-button").simulate('click', () => {
        expect(mockhandleLogin).toHaveBeenCalled()
      })
    })

    it('should call handlelogin when button is clicked', () => {
      wrapper.find("form").simulate('click', () => {
        expect(mockhandleLogin).toHaveBeenCalled()
      })
    })

    it.skip('responds to name change', () => {
      const event = {target: {name: 'password', value: 'cool'}}; 
      wrapper.find('#emails').simulate('change', event);
      expect(mockChange).toEqual('cool');
    });  
})