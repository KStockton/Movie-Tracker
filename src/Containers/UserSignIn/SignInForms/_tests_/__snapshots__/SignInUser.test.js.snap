// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewUser NewUser component should match the snapshot 1`] = `
<section
  className="user-sign-in-container"
>
  <div
    className="user-sign-in-background"
  >
    <img
      alt="movies"
      src="https://i.pinimg.com/originals/86/f1/99/86f199d0c6f5cedd14c121164fa5fafc.jpg"
    />
  </div>
  <form
    className="sign-in-form login-form"
    onSubmit={[MockFunction]}
  >
    <h3
      className="user-form-title"
    >
      Sign In
    </h3>
    <div
      className="input-container"
    >
      <input
        className="user-input"
        id="emails"
        name="email"
        onChange="cool"
        placeholder="Email..."
        type="email"
      />
    </div>
    <div
      className="input-container"
    >
      <input
        className="user-input"
        name="password"
        onChange="cool"
        placeholder="Password..."
        type="password"
      />
    </div>
    <span
      className="status-msg"
    />
    <div
      className="button-container"
    >
      <button
        className="user-form-button sign-in-button"
        onClick={[MockFunction]}
      >
        Sign In
      </button>
      <button
        className="user-form-link"
        onClick={[MockFunction]}
      >
        New user? Create an Account!
      </button>
    </div>
  </form>
</section>
`;
